File: Python/Problem2.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        patches=0
        i=0
        sumatory=0
        length=len(nums)
        while sumatory<n:
            if i<length and nums[i]<=sumatory+1:
                sumatory+=nums[i]
                i+=1
            else:
                patches+=1
                sumatory+=(sumatory+1)
        return patches

File: Java/Problem2.java
class Solution {
    public int minPatches(int[] nums, int n) {
        int patches=0;
        int i=0;
        long sumatory=0;
        int length= nums.length;
        while (sumatory<n) {
            if (i<length && nums[i]<=sumatory+1){
                sumatory+= nums[i];
                i++;
            } else {
                patches++;
                sumatory+=(sumatory+1);
            }
        }
        return patches;
    }
}